src/integer.rs

```diff
@@ -91,6 +91,29 @@ impl Int {
     }
 }
 
+impl<'a> Neg for &'a Int {
+    type Output = Int;
+
+    fn neg(self) -> Int {
+        if *self == Int::from(0) {
+            Int::from(0)
+        } else {
+            Int {
+                is_negative: !self.is_negative,
+                digits: self.digits.clone(),
+            }
+        }
+    }
+}
+
+impl Neg for Int {
+    type Output = Int;
+
+    fn neg(self) -> Int {
+        -&self
```

Comment:
I realized we should just be modifying self directly. neg(self) takes ownership of self here so it should just modify it and return it. By calling neg(&self) we make an unnecessary copy.